sql/updates/README.md

```diff
@@ -1,3 +1,32 @@
+## General principles for statements in update/downgrade scripts
+
+1. The `search_path` for these scripts will be locked down to
+  `pg_catalog`. Locking down `search_path` happens in `pre-update.sql`.
+  Therefore all object references need to be fully qualified unless
+  they reference objects from `pg_catalog`. Use `@extschema@` to refer
+  to the target schema of the installation (resolves to `public` by
+  default).
+2. Creating objects must not use IF NOT EXISTS as this will
+  introduce privilege escalation vulnerabilities.
+3. All functions should have explicit `search_path`. Setting explicit
+  `search_path` will prevent SQL function inlining for functions and
+  transaction control for procedures so for some functions/procedures
+  it is acceptable to not have explicit `search_path`. Special care
+  needs to be taken with those functions/procedures by either setting
+  `search_path` in function body or having only fully qualified object
+  references including operators.
+4. When generating the install scripts `CREATE OR REPLACE` will be
```

Comment:
Are the install scripts not run during the updates as well?

Reply:
Anything in sql/pre_install (except types.functions.sql and fdw_functions.sql) is not run during updates.